src/views/Upload.vue

```diff
@@ -36,7 +36,26 @@
 					@text-change="updateText"
 				/>
 			</div>
-			<div class="flex flex-row justify-end items-end text-center w-full px-2 md:px-20 md:text-right pb-2 md:pb-10 mt-10 md:mt-10 ">
+
+			<div class="flex flex-row justify-end items-end text-center w-full px-2 md:px-20 md:text-right pb-2 md:pb-10 mt-10 md:mt-10 ">		
+				<div class="flex flex-row">	
+					<input 
+						v-model="checked" 
+						type="checkbox" 
+						class="m-2"
+					>
+					<div v-if="checked == true" class="flex-row pr-5">
+						<input 
+							v-model="message" 
```

Comment:
Please add some layer of validation before sending to backend - if the checkbox is enabled, but the mail isn't provided or is empty string, throw an error or sth.